docs/usage/customization-guide.md

```diff
@@ -594,6 +594,49 @@ details.
 > labels specified in the `labels` field will override anything
 > originating from `labelsTemplate`.
 
+#### Node Annotations
+
+The `.annotations` field is a list of features to be advertised as annotations.
+
+Take this rule as a referential example:
+
+```yaml
+apiVersion: nfd.k8s-sigs.io/v1alpha1
+kind: NodeFeatureRule
+metadata:
+  name: feature-annotations-example
+spec:
+  rules:
+    - name: "annotation-example"
+      annotations:
+        defaul-ns-annotation: "foo"
+        feature.node.kubernetes.io/defaul-ns-annotation-2: "bar"
+        custom.vendor.io/feature: "baz"
+        kubernetes.io/feature: "denied"
+        subns.kubernetes.io/blah: "denied"
+        nfd.node.kubernetes.io/xyz: "denied"
```

Comment:
I think it would probably be best just to drop these "illegal" annotations from the example.

However, we should add a note about the ns (or prefix/) restrictions on annotation names. Similar what we have for labels, ERs and taints (you can look at an example text from there).